docs/articles/documentation/test-api/selecting-page-elements/examples-of-working-with-dom-elements.md

```diff
@@ -81,7 +102,48 @@ test('My test', async t => {
 });
 ```
 
-## Checking if an Element is Available
+## Access Child Nodes in the DOM Hierarchy
+
+The selector API allows you to [search for elements in the DOM tree](selectors/functional-style-selectors.md#search-for-elements-in-the-dom-hierarchy). You can traverse in different directions through the hierarchy to iterate parents, children and siblings.
+
+However, the [child](selectors/functional-style-selectors.md#child), [sibling](selectors/functional-style-selectors.md#sibling), [prevSibling](selectors/functional-style-selectors.md#prevsibling) and [nextSibling](selectors/functional-style-selectors.md#nextsibling) methods work with DOM elements only and ignore other node types.
+
+This example shows how to get a child text node for a given parent element with client-side code. Consider the following *example.html* page:
+
+```js
+<!DOCTYPE html>
+<html>
+    <body id="tested-page">
+        This is my tested page. <!--This is the first child node of <body>-->
+        <p>My first paragraph.</p>
+        <p>My second paragraph.</p>
+    </body>
+</html>
+```
+
+Assume you have a selector that identifies the `#tested-page` element, and now you need to verify text content of its first child node (*'This is my tested page'*).
+
+To do this, pass this selector as a [dependency](selectors/selector-options.md#optionsdependencies) to a new selector. Initialize the second selector with a function that returns the [childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes) collection. You can then use the new selector in an assertion to check the `textContent` property.
+
+```js
+import { Selector, ClientFunction } from 'testcafe';
+
+fixture `My Fixture`
+    .page `example.html`;
+
+const testedPage = Selector('#tested-page');
+
+const childNodes = Selector(() => {
+    return getPage().childNodes;
+},{ dependencies: { getPage: testedPage } });
+
+
+test('My Test', async t => {
+    await t.expect(childNodes.nth(0).textContent).contains('This is my tested page.');
```

Comment:
I'm not sure that it's a good assertion because even if we fail finding the node this assertion passes.
```js
 test('My Test', async t => {
    await t.expect(testedPage.nth(0).textContent).contains('This is my tested page.');
});
```

Reply:
What about methods that can search through nodes (see my [comment](https://github.com/DevExpress/testcafe/pull/3855#discussion_r289348163))
For this example it probably would be easier to use the `find` method, don't you think
```js
const firstChildNode = Selector('#tested-page').find((node, index) => {
    return index === 0;
});


test('My Test', async t => {
    await t.expect(firstChildNode.textContent).eql('\nThis is my tested page. ');
});
```

Reply:
Maybe we can use a more real-wordish example here? 

https://github.com/DevExpress/testcafe/issues/1053#issuecomment-266721888

```js
import { Selector } from 'testcafe';

fixture `My fixture`
    .page `https://chrisbateman.github.io/guide-to-web-components/demos/shadow-dom.htm`;

const demoPage = Selector('#demo1');

const paragraph = Selector(() => {
    return demoPageSelector().shadowRoot.querySelectorAll('p');
}, { dependencies: { demoPageSelector: demoPage } });

test('Get text within shadowroot', async t => {    
    await t.click(paragraph.nth(0));
    
    var text = await paragraph.nth(0).textContent;

    await t.expect(paragraph.nth(0).textContent).eql('These paragraphs are in a shadow root.');
});
```